src/libraries/Native/Unix/System.Globalization.Native/pal_icushim.c

```diff
@@ -306,41 +366,36 @@ static int FindICULibs(const char* versionPrefix, char* symbolName, char* symbol
 
 #endif
 
+void ValidateICUDataCanLoad()
+{
+    UVersionInfo version;
+    UErrorCode err = U_ZERO_ERROR;
+    ulocdata_getCLDRVersion(version, &err);
+
+    if (U_FAILURE(err))
+    {
+        fprintf(stderr, "Could not load ICU data. UErrorCode: %d\n", err);
```

Comment:
@tarekgh @jkotas if we're not able to load ICU data in Windows (not app-local), should we fail fast or fall back to NLS?

Reply:
what is the possibility this can happen?

Reply:
AFAICT (based on past experience), this would really only happen if:
- You're already pretty close to exhausting the memory for the system, or your job/process is near the limit/quota, so loading the data file would cause OOM error to happen.
- Or the ICU data file (`%WINDIR%\Globalization\icudtl.dat`) is corrupted, missing, or otherwise bad.

Both are indeed quite rare. Though at least with ChakraCore the first one did happen on rare occasions, as IIRC, I think they ran each tab in a separate job/process that had rather strict memory limits. (So hitting an OOM would happen every so often).

Reply:
Sounds good, thanks for the info.